platforms/jvm/plugins-application/src/main/java/org/gradle/jvm/application/scripts/TemplateBasedScriptGenerator.java

```diff
@@ -37,6 +38,7 @@ public interface TemplateBasedScriptGenerator extends ScriptGenerator {
      * @return Template reader
      */
     @ToBeReplacedByLazyProperty
+    @InputFile
```

Comment:
I think this needs to be `@Nested` too

Reply:
Thanks for the review! I have updated the commit, and also replace `@Internal` with `@Nested` on `getWindowsStartScriptGenerator` in `CreateStartScripts.java`.

But I'm still not sure what to do with the failing test `ProviderMigrationArchitectureTest. public_api_properties_should_not_have_migration_annotation`.